src/Cli.Tests/EndToEndTests.cs

```diff
@@ -274,102 +294,96 @@ public void TestConfigGeneratedAfterAddingEntityWithSourceAsStoredProcedure()
     [TestMethod]
     public void TestConfigGeneratedAfterUpdatingEntityWithSourceAsStoredProcedure()
     {
-        string? runtimeConfigJson = AddPropertiesToJson(INITIAL_CONFIG, SINGLE_ENTITY_WITH_STORED_PROCEDURE);
-        WriteJsonContentToFile(TEST_RUNTIME_CONFIG_FILE, runtimeConfigJson);
-        RuntimeConfig? runtimeConfig = TryGetRuntimeConfig(TEST_RUNTIME_CONFIG_FILE);
-        Assert.IsNotNull(runtimeConfig);
-        string expectedSourceObject = @"{
-            ""type"": ""stored-procedure"",
-            ""object"": ""s001.book"",
-            ""parameters"": {
-                ""param1"": 123,
-                ""param2"": ""hello"",
-                ""param3"": true
-            }
-        }";
-
-        string actualSourceObject = JsonSerializer.Serialize(runtimeConfig.Entities["MyEntity"].Source);
-        Assert.IsTrue(JToken.DeepEquals(JObject.Parse(expectedSourceObject), JObject.Parse(actualSourceObject)));
+        string runtimeConfigJson = AddPropertiesToJson(INITIAL_CONFIG, SINGLE_ENTITY_WITH_STORED_PROCEDURE);
+
+        _fileSystem!.File.WriteAllText(TEST_RUNTIME_CONFIG_FILE, runtimeConfigJson);
 
         // args for update command to update the source name from "s001.book" to "dbo.books"
         string[] updateArgs = { "update", "MyEntity", "-c", TEST_RUNTIME_CONFIG_FILE, "--source", "dbo.books" };
-        Program.Main(updateArgs);
-        runtimeConfig = TryGetRuntimeConfig(TEST_RUNTIME_CONFIG_FILE);
+        Program.Execute(updateArgs, _cliLogger!, _fileSystem!, _runtimeConfigLoader!);
+
+        Assert.IsTrue(_runtimeConfigLoader!.TryLoadConfig(TEST_RUNTIME_CONFIG_FILE, out RuntimeConfig? runtimeConfig));
         Assert.IsNotNull(runtimeConfig);
-        expectedSourceObject = @"{
-            ""type"": ""stored-procedure"",
-            ""object"": ""dbo.books"",
-            ""parameters"": {
-                ""param1"": 123,
-                ""param2"": ""hello"",
-                ""param3"": true
-            }
-        }";
-
-        actualSourceObject = JsonSerializer.Serialize(runtimeConfig.Entities["MyEntity"].Source);
-        Assert.IsTrue(JToken.DeepEquals(JObject.Parse(expectedSourceObject), JObject.Parse(actualSourceObject)));
+        Entity entity = runtimeConfig.Entities["MyEntity"];
+        Assert.AreEqual(EntitySourceType.StoredProcedure, entity.Source.Type);
+        Assert.AreEqual("dbo.books", entity.Source.Object);
+        Assert.IsNotNull(entity.Source.Parameters);
+        Assert.AreEqual(3, entity.Source.Parameters.Count);
+        Assert.AreEqual(123, entity.Source.Parameters["param1"]);
+        Assert.AreEqual("hello", entity.Source.Parameters["param2"]);
+        Assert.AreEqual(true, entity.Source.Parameters["param3"]);
```

Comment:
Is it not possible to use `Verify()` here like the test above this one where `_runtimeConfigLoader!.TryLoadConfig(TEST_RUNTIME_CONFIG_FILE, out RuntimeConfig? runtimeConfig)` is used to get a before and after version of the file? looks like that can happen berfore the first Program.Execute usage.